### PR TITLE
Add the ability to disable ToolbarActions #2306

### LINKED ISSUE #2306: Add the ability to disable ToolbarActions
ToolbarActions and MenuItems needs to be disableable for the similar reasons and coherence with DisableableWidgets. Last corresponds to solved #234 .

### PR DESCRIPTION
**Pull request by pravin772**
### Description:
Issue addressed Add the ability to disable ToolbarActions #2306
The change is inspired from same properties as Button for ToolbarAction.
Comments from issue were helpful and followed during code change.

Fixes #2306 

### Checklist:

- [x] Tests included.
- [ ] Lint and formatter run with no errors.
- [x] Tests all pass.

### CODE REVIEW

**Comment by pravin772** [comment]
Hi,
Seen this issue open and labeled "good first issue". So, tried to implement this feature. Happy to contribute more.
Please let me know if any change necessary. Happy to comply with required changes. Thanks

File: widget/toolbar.go [hunk 1]
```diff
@@ -37,7 +38,17 @@ func (t *ToolbarAction) SetIcon(icon fyne.Resource) {
...
+		Icon:        icon,
+		OnActivated: onActivated,
+	}
+	return toolbarAction
+}
+
+// SetDisabled updated the disabled property of ToolbarAction
+func (t *ToolbarAction) SetDisabled(disabled bool) {
+	t.disabled = disabled
+	t.ToolbarObject().Refresh()
```
Comment (andydotxyz) [inline widget/toolbar.go:51]: This doesn't seem right - it creates a new object visual and then refreshes it instead of refreshing the current UI.

**Review by andydotxyz — CHANGES_REQUESTED** [review]
Unfortunately embedding DisablableWidget is not the right thing to do, as these are not widgets.

Matching the functions it provides does make sense though.

File: widget/toolbar.go (hunk 1 above)
Comment (pravin772) [reply widget/toolbar.go:51]: Thanks for review. Does t.Refresh() makes sense? That will re-render same element.

**Comment by pravin772** [comment]
> Unfortunately embedding DisablableWidget is not the right thing to do, as these are not widgets.
> 
> Matching the functions it provides does make sense though.

Do you reckon should we use separate disabled property in ToolbarAction struct?

**Comment by andydotxyz** [comment]
Yes I think so - replicating DisabledWidget without bringing in BaseWidget along with it

**Comment by pravin772** [comment]
Added disabled property with all equivalent methods as DisabledWidget. Could you suggest how can I refresh that element.
Cannot use t.Refresh() as it is no more BaseWidget. Moreover, do I need to update styles specific for disabled ToolbarAction?

**Review by andydotxyz — CHANGES_REQUESTED** [review]
This looks clean, but it does not seem to refresh the widget when `disabled` changes.

**Comment by pravin772** [comment]
Hi, I was not able to use t.Refresh() as it is not BaseWidget anymore. However, I found that t.ToolbarObject().Refresh() can be used as t.ToolbarObject() gets a button to render this ToolbarAction. Let me know if that works otherwise please could you suggest any solution. Thanks

**Comment by andydotxyz** [comment]
> Hi, I was not able to use t.Refresh() as it is not BaseWidget anymore. However, I found that t.ToolbarObject().Refresh() can be used as t.ToolbarObject() gets a button to render this ToolbarAction. Let me know if that works otherwise please could you suggest any solution. Thanks

Indeed, you'll have to find the right thing to refresh. Unfortunately `ToolbarObject()` creates a new instance, so you are refreshing something not on screen.
If you test this in a real app you should notice that it does not update if you toggle it after drawn for the first time

**Pull request closed without merging**